### PR TITLE
Fix PHP 7.3 deprecations in lib/ directory (Part 1).

### PR DESCRIPTION
**Pull request by syntaxseed**
Have begun fixing PHP 7.3->7.4 deprecations for the project. Began with lib/ directory.

Fixed:
- Curly brace syntax for accessing array elements and string offsets has been deprecated in PHP 7.4.
- Passing the $glue and $pieces parameters in reverse order to join has been deprecated since PHP 7.4; $glue should be the first parameter and $pieces the second.

I will continue this work in future PRs. Just making sure this is wanted. :)

Using PHPCS:
```bash
phpcs --standard=PHPCompatibility --extensions=php,inc,lib --runtime-set testVersion 7.3- . 
```

### CODE REVIEW

**Comment by phy25** [comment]
My apologies but I have done something here:
https://github.com/splitbrain/dokuwiki/tree/fix-tests-php74

Not opening a new PR for this yet because we are waiting for #2850 (and after this get merged a lot of PRs needs to be rebased in order to get the tests working).

**Comment by splitbrain** [comment]
@phy25 I'm unsure about the status here. Does your branch replace this PR? I'd be happy to have this (or something based on this) merged.

**Comment by syntaxseed** [comment]
I can rebase if needed.

**Comment by phy25** [comment]
My branch, notably 439cabd71dd05b39ec44b270ee099c27e2b844bc, is a superset of this one. I guess we can merge this one first and I can rebase mine.

**Comment by syntaxseed** [comment]
@phy25 sounds good!

**Comment by phy25** [comment]
I finally have some time to look into this. This (and mine) needs to be worked again from the beginning because of massive conflicts.

I will keep this open and delete my commits regarding this.

**Comment by syntaxseed** [comment]
@phy25 Ok so want me to rebase this and start again?

**Comment by phy25** [comment]
@syntaxseed Yes you are welcome to do so.

**Comment by syntaxseed** [comment]
Please bear with me as this is my first large rebase on a forked project. I think I have rebased successfully onto the latest from master and updated my PR.

File: lib/plugins/info/syntax.php [hunk 1]
```diff
@@ -189,7 +189,7 @@ protected function renderHelperMethods(Doku_Renderer_xhtml $renderer)
                     foreach ($method['params'] as $desc => $type) {
                         $params[] = hsc($desc).'</td><td>'.hsc($type);
                     }
-                    $doc .= join($params, '</td></tr><tr><td>').'</td></tr>';
+                    $doc .= join('</td></tr><tr><td>', $params).'</td></tr>';
```
Comment (phy25) [inline lib/plugins/info/syntax.php]: Good catch!

**Pull request merged**